File: src/JS/browserAction.js
function setBadgeText(text) {
    chrome.browserAction.setBadgeText({ text: text });
}

function getTabUrl() {
    return new Promise((resolve, reject) => {
        chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
            let url = tabs[0].url;
            resolve(url);
        });
    });
}

export { setBadgeText, getTabUrl };

File: src/JS/BrowserAction.fs
namespace JS

open Fable.Core

type IBrowserAction =
    abstract setBadgeText: string -> unit
    abstract getTabUrl: unit -> JS.Promise<string>

module BrowserAction =

    [<Import("*", "./browserAction.js")>]
    let private native: IBrowserAction = jsNative

    let setBadgeText = native.setBadgeText
    let getTabUrl = native.getTabUrl
